week-005-array-min-swap-2/johee.js

```diff
@@ -1,23 +1,35 @@
 function minimumSwaps(arr) {
+    // creating an object. Can edit objects with a const as long as I don't reassign the variable. Fun
     const touchedNodes = {};
+    // using let to initialize reassignable variables. New declaration keyword helps with scope
     let currentSwaps = 0;
     let totalSwaps = 0;
     
     for (let i = 0; i < arr.length; ++i) {
+        // if the node was touched, it's already in the right spot
         if (touchedNodes[i]) {
             continue;
         }
         
+        // well, let's "touch" (check) the node to see if it's in the right spot
         touchedNodes[i] = true;
```

Comment:
...what..? how does =true; mean it's in the right spot?

Reply:
what javascript wizardy is this?!

Reply:
It just means that I am going to check it